soc/ti/k3/j721e/Kconfig.soc

```diff
@@ -0,0 +1,24 @@
+# Copyright (C) 2023 BeagleBoard.org Foundation
+# Copyright (C) 2023 S Prashanth
+#
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_SERIES_J721E
```

Comment:
It looks like there are more SoCs in this series (e.g. J7200), consider renaming to `SOC_SERIES_J7X` or similar

Reply:
I agree, this should be more generic like we did for the am6x.

Reply:
I'm okay with more generic, but I don't think giving it a more generic names helps us here.

TI's K3 processor line naming scheme doesn't tend to work well with generic names and wildcards like "x". For instance AM62P and AM62A are very different parts, and there is also a AM62 part which is very different still from the other two. So AM62x would be an incorrect "generic" as it would seem to support 3 parts, which definitely do not have common M4 Zephyr support.

I'd like to do what DT folks do with compatible strings: use the full name of the first device that is compatible, then all later compatible devices simply re-use that name. Only when an incompatible device comes out do we add the next name. So here, J721e is the first part with R5 cores, J7200, J784s4, etc, would simply make use of it. Then J722s, which has different R5 support, would simply start the new line for which follow up compatible devices then use.

Reply:
That's a great point, let's wait for the next SoC(s) and refactor if/when it makes sense

Reply:
Can you pick the right name now? It's better to have a soc named correctly at the start than to add it with a wrong name then break it entirely for any out of tree users that have started using it when they update after you've fixed it